src/core/jwk.rs

```diff
@@ -402,6 +437,120 @@ impl
     }
 }
 
+///
+/// Variants of EdDSA Key.
+///
+pub enum CoreEdDsaKeyVariant {
+    ///
+    /// Ed25519 Variant
+    /// 
+    Ed25519,
+}
+
+struct EdDsaSigningKey {
+    variant: CoreEdDsaKeyVariant,
+    ed25519: Option<ed25519_dalek::SigningKey>,
+}
+
+impl EdDsaSigningKey {
+    fn from_pem(pem: &str, variant: CoreEdDsaKeyVariant) -> Result<Self, String> {
+        match variant {
+            CoreEdDsaKeyVariant::Ed25519 => Ok(Self {
+                variant,
+                ed25519: Some(ed25519_dalek::SigningKey::from_pkcs8_pem(pem).map_err(|err| err.to_string())?),
+            })
+        }
+    }
+
+    fn public_part(&self) -> Vec<u8> {
+        match self.variant {
+            CoreEdDsaKeyVariant::Ed25519 => {
+                self.ed25519
+                    .as_ref()
+                    .expect("fail to refer to Ed25519 signing key")
+                    .verifying_key()
+                    .as_bytes()
+                    .to_vec()
+            }
+        }
+    }
+
+    fn sign(&self, message: &[u8]) -> Vec<u8> {
+        match self.variant {
+            CoreEdDsaKeyVariant::Ed25519 => {
+                let key = self.ed25519.as_ref().expect("fail to refer to Ed25519 signing key");
+
+                let signature = key.sign(message);
+
+                signature.to_vec()
+            }
+        }
+    }
+}
+
+///
+/// EdDSA Private Key.
+///
+/// This key can be used for signing messages, or converted to a `CoreJsonWebKey` for verifying
+/// them.
+///
+pub struct CoreEdDsaPrivateSigningKey {
+    kid: Option<JsonWebKeyId>,
+    key_pair: EdDsaSigningKey,
+}
+impl CoreEdDsaPrivateSigningKey {
+    ///
+    /// Converts an EdDSA private key (in PEM format) to a JWK representing its public key.
+    ///
+    pub fn from_pem(pem: &str, kid: Option<JsonWebKeyId>, variant: CoreEdDsaKeyVariant) -> Result<Self, String> {
+        match variant {
+            CoreEdDsaKeyVariant::Ed25519 => Ok(Self {
+                kid,
+                key_pair: EdDsaSigningKey::from_pem(pem, variant)?,
+            })
+        }
+    }
+}
+impl
+    PrivateSigningKey<
+    CoreJwsSigningAlgorithm,
+    CoreJsonWebKeyType,
+    CoreJsonWebKeyUse,
+    CoreJsonWebKey,
+    > for CoreEdDsaPrivateSigningKey
+{
+    fn sign(&self, signature_alg: &CoreJwsSigningAlgorithm, message: &[u8]) -> Result<Vec<u8>, SigningError> {
```

Comment:
accepting a `signature_alg` here means we could (in the future) end up with a conflict between the key type (e.g., `EdDsaSigningKey::Ed25519`) and the `signature_alg` (e.g., Ed448). since, for EdDSA, the algorithm seems to be determined exclusively by the key type (vs. RSA that has different hash functions available to use with a given key), I think we should just remove the `signature_alg` argument. this also means the function will become infallible, and we no longer need to worry about an error type.

Reply:
hello @ramosbugs, thanks for your reviews, you mean ignoring the argument right? since the `PrivateSigningKey` trait has the `signature_alg` parameter?

Reply:
oh! I missed that this was a trait impl 🤦‍♂️

nevermind, ignore my comment :-)